HOSTEDCP-1184: Document IPv6/IPv4/DualStack deployments for Hypershift in Baremetal (#3008)
openshift/hypershift: docs/content/how-to/mce/Dual/hostedcluster/baremetalhost.md

Reviewer: This file shares a lot in common with the existing https://hypershift-docs.netlify.app/how-to/agent/create-agent-cluster/. Perhaps these two can be reconciled together or this could replace the other one?

Author: humm I don't think to replace it is a good choice, but I've remapped  the location for this documentation and I think is much better now. Let me know what's you opinion on this :)

  namespace: clusters-hosted-dual
data:
  password: YWRtaW4=
  username: YWRtaW4=
type: Opaque
---
apiVersion: metal3.io/v1alpha1
kind: BareMetalHost
metadata:
  name: hosted-dual-worker0
  namespace: clusters-hosted-dual
  labels:
    infraenvs.agent-install.openshift.io: hosted-dual
  annotations:
    inspect.metal3.io: disabled
    bmac.agent-install.openshift.io/hostname: hosted-dual-worker0
spec:
  automatedCleaningMode: disabled
  bmc:
    disableCertificateVerification: true
    address: redfish-virtualmedia://[192.168.126.1]:9000/redfish/v1/Systems/local/hosted-dual-worker0
    credentialsName: hosted-dual-worker0-bmc-secret
  bootMACAddress: aa:aa:aa:aa:02:11
  online: true
```

**Details**:

- We will have at least 1 secret which holds the BMH credentials, so we will need to create at least 2 objects per worker node.
- `spec.metadata.labels["infraenvs.agent-install.openshift.io"]` Is the link between the Assisted Installer and the BareMetalHost objects.
- `spec.metadata.annotations["bmac.agent-install.openshift.io/hostname"]` Is the node name it will take on deployment.
- `spec.automatedCleaningMode` prevents the node to be erased by the Metal3 operator.

> - `spec.bmc.disableCertificateVerification` Is set to `true`` avoid the certification validation from the client.

Author: I've modified the style a bit, I think is better now.

```
NAMESPACE         NAME                                   CLUSTER   APPROVED   ROLE          STAGE
clusters-hosted   aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaa0411             true       auto-assign
clusters-hosted   aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaa0412             true       auto-assign
clusters-hosted   aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaa0413             true       auto-assign
```

These agents represent the nodes that are available for an installation. To assign them to a HostedCluster we need to scale up the NodePool.

## Scaling up the Nodepool

Once we have the BareMetalHosts created, the statuses of these BareMetalHost will transition from `Registering` (Trying to reach the Node's BMC) to `Provisioning` (Booting up the node) and finally to `Provisioned` (The node has been booted properly).

The nodes will boot with the Agent's RHCOS LiveISO and with a pod running by default called "agent". This agent is the piece in the node that will receive the steps from the Assisted Service Operator to install Openshift payload.

To do so we just need to execute this command:

```bash
oc -n clusters scale nodepool hosted-dual --replicas 3
```

> After the nodepool scalation we can see the agents are assigned to a Hosted Cluster

Reviewer: ```suggestion
After the nodepool escalation, we can see the agents are assigned to a Hosted Cluster
```

```
NAMESPACE         NAME                                   CLUSTER   APPROVED   ROLE          STAGE
clusters-hosted   aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaa0411   hosted    true       auto-assign
clusters-hosted   aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaa0412   hosted    true       auto-assign
clusters-hosted   aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaa0413   hosted    true       auto-assign
```

And the NodePool replicas set

```
NAMESPACE   NAME     CLUSTER   DESIRED NODES   CURRENT NODES   AUTOSCALING   AUTOREPAIR   VERSION                              UPDATINGVERSION   UPDATINGCONFIG   MESSAGE
clusters    hosted   hosted    3                               False         False        4.14.0-0.nightly-2023-08-29-102237                                      Minimum availability requires 3 replicas, current 0 available
```

> So now we need to wait until the Nodes join the cluster. The Agents will give you some hints about in which stage they are and what is the status. Initially they does not post any status, but eventually they will.

Reviewer: ```suggestion
So, now we need to wait until the Nodes join the cluster. The Agents will give you some hints about which stage they are and what is their status. Initially, they do not post any status, but eventually they will.
```
